### PR TITLE
DDP-6752: *NEW* [Pancan] picklist autocomplete : treat the "-", '/' characters as a space.

### PR DESCRIPTION
**Pull request by alextanlan**
Another implementation approach.
Also fixed needed comments suggested in the [previous PR#842](https://github.com/broadinstitute/ddp-angular/pull/842)

The ticket: [DDP-6752](https://broadinstitute.atlassian.net/browse/DDP-6752)
Also see [the slack discussion](https://broadinstitute.slack.com/archives/C02AAP2AM98/p1628262486037600).

### CODE REVIEW

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts [hunk 1]
```diff
@@ -56,43 +58,33 @@ export class AutocompleteActivityPicklistQuestion extends BaseActivityPicklistQu
...
+        const sortedPicklistGroups = this.sortPolicy.sortPicklistGroups(this.block.picklistGroups);
+        const sortedPicklistOptions = this.sortPolicy.sortPicklistOptions(this.block.picklistOptions);
 
-        userQueryStream.pipe(startWith('')).subscribe((value: string | ActivityPicklistOption) => {
-            const query = typeof value === 'string' ? value : value.optionLabel;
-            this.filteredGroups = this.filterOutEmptyGroups(query ? this.filterGroups(query, sortedPicklistGroups)
-                : sortedPicklistGroups.slice());
-            this.filteredOptions = query ? this.filterOptions(query, sortedPicklistOptions) : sortedPicklistOptions.slice();
-        });
+        userQueryStream$.pipe(startWith(''))
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts:79]: Only changed `userQueryStream` => `userQueryStream$` here and aligned the formatting.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts [hunk 2]
```diff
@@ -56,43 +58,33 @@ export class AutocompleteActivityPicklistQuestion extends BaseActivityPicklistQu
...
-        if (answer) {
-            const value = answer.detail || [
-                ...this.block.picklistGroups.flatMap(group => group.options),
-                ...this.block.picklistOptions
-            ].find(option => option.stableId === answer.stableId);
-            this.inputFormControl.setValue(value);
-        }
+        // reset the sort policy for picklists which don't need to have sorted options
+        this.sortPolicy = this.shouldBeSorted ? this.sortPolicy : new PicklistSortingPolicy();
+        this.initInputValue();
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts:67]: extracted lines 62-69 into a separate method.

File: ddp-workspace/projects/ddp-sdk/src/lib/utility/stringsHelper.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export class StringsHelper {
+    static normalizeString(text: string, ignoredSymbols: string[], separator: string = ' '): string {
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/utility/stringsHelper.ts:2]: Replacing all symbols to be ignored with a space.
Case-insensitive.

**Review by MocanaAtBroad — APPROVED** [review]
Looks good! ✅

**Pull request merged**